packages-exp/performance-exp/src/services/transport_service.ts

```diff
@@ -90,9 +91,11 @@ function processQueue(timeOffset: number): void {
 }
 
 function dispatchQueueEvents(): void {
-  // Capture a snapshot of the queue and empty the "official queue".
-  const staged = [...queue];
-  queue = [];
+  // Extract events up to the maximum cap of single logRequest from top of "official queue".
+  // The staged events will be used for current logRequest attempt, remaining events will be kept
+  // for next attempt.
+  const staged = queue.slice(0, MAX_EVENT_COUNT_PER_REQUEST);
```

Comment:
use splice instead.

Reply:
Good suggestion! Applied `splice()` to split the array.